### PR TITLE
[Resolver]: Indicating more informative failures

### PR DESCRIPTION
**Pull request by malhotra5**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

When the resolver is invoked on an issue, it handles the outcome in one of two ways

1. Successful: leave link to a PR in a comment on the issue
2. Unsuccessful: leave link to a branch in a comment on the issue

This PR adds additional failure explanations in the comment body when the resolver is unsuccessful. Note, we don't do this for the successful case, as the success explanation is available in the PR body. 

---
**Link of any specific issues this addresses**
#5137

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:38808b8-nikolaik   --name openhands-app-38808b8   docker.all-hands.dev/all-hands-ai/openhands:38808b8
```

### CODE REVIEW

File: .github/workflows/openhands-resolver.yml [hunk 1]
```diff
@@ -340,11 +359,17 @@ jobs:
               });
               process.env.AGENT_RESPONDED = 'true';
             } else if (!success && branchName) {
+              let commentBody = `An attempt was made to automatically fix this issue, but it was unsuccessful. A branch named '${branchName}' has been created with the attempted changes. You can view the branch [here](https://github.com/${context.repo.owner}/${context.repo.repo}/tree/${branchName}). Manual intervention may be required.`;
+
+              if (successExplanation) {
+                commentBody += `\n\nAdditional details about the failure:\n${successExplanation}`;
```
Comment (enyst) [inline .github/workflows/openhands-resolver.yml:365]: Just nit, this line reads amusing. The failure is the success explanation 😅

If `successExplanation` can be both about success and failure, could we call it... idk, `attemptExplanation`?

Comment (enyst) [reply .github/workflows/openhands-resolver.yml:365]: Or `resultExplanation`?

Comment (malhotra5) [reply .github/workflows/openhands-resolver.yml:365]: Yeah that's a good point! I'll probably refactor that in a separate PR

Comment (tofarr) [reply .github/workflows/openhands-resolver.yml:365]: Maybe just rename the variable from `successExplanation` to `resultExplanation`?

Comment (malhotra5) [reply .github/workflows/openhands-resolver.yml:365]: I also want to make these changes across the resolver codebase in addition to the workflow definition

I'll get Openhands to do this for me in a separate PR

Comment (enyst) [reply .github/workflows/openhands-resolver.yml:365]: I agree it makes sense to do the full renaming, I suspected it's in more places than just this (I must have seen it elsewhere in passing).

This isn't solved though, so I reopened it. 😅

Comment (malhotra5) [reply .github/workflows/openhands-resolver.yml:365]: Ah right, thanks a bunch!

**Pull request merged**